src/main/java/seedu/us/among/model/endpoint/Endpoint.java

```diff
@@ -75,8 +75,18 @@ public boolean isSameEndpoint(Endpoint otherEndpoint) {
         if (otherEndpoint == this) {
             return true;
         }
-        return otherEndpoint != null && otherEndpoint.getMethod().equals(getMethod())
-                && otherEndpoint.getAddress().equals(getAddress());
+
+        //expanded this for more clarity
+        if (otherEndpoint == null) {
+            return false;
+        } else {
+            if (otherEndpoint.getMethod().equals(getMethod()) && otherEndpoint.getAddress().equals(getAddress())
+                    && otherEndpoint.getTags().equals(getTags())) {
+                return true;
+            } else {
+                return false;
+            }
+        }
```

Comment:
You can actually use the Equals method in this same class since the equals method checks for if method, address and tags are the same.